## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 477891`
Commit: `6d065b71c92bd8d4c9c9a5215f3cdf1a541d6bab`

---
### `x86_64-linux`
<details>
  <summary>:x: 24 packages failed to build:</summary>
  <ul>
    <li>python314Packages.coiled</li>
    <li>python314Packages.coiled.dist</li>
    <li>python314Packages.compressai</li>
    <li>python314Packages.compressai.dist</li>
    <li>python314Packages.heudiconv</li>
    <li>python314Packages.heudiconv.dist</li>
    <li>python314Packages.nipype</li>
    <li>python314Packages.nipype.dist</li>
    <li>python314Packages.niworkflows</li>
    <li>python314Packages.niworkflows.dist</li>
    <li>python314Packages.pybids</li>
    <li>python314Packages.pybids.dist</li>
    <li>python314Packages.skytemple-dtef</li>
    <li>python314Packages.skytemple-dtef.dist</li>
    <li>python314Packages.skytemple-files</li>
    <li>python314Packages.skytemple-files.dist</li>
    <li>python314Packages.skytemple-ssb-debugger</li>
    <li>python314Packages.skytemple-ssb-debugger.dist</li>
    <li>python314Packages.spacy-loggers</li>
    <li>python314Packages.spacy-loggers.dist</li>
    <li>python314Packages.templateflow</li>
    <li>python314Packages.templateflow.dist</li>
    <li>python314Packages.wandb</li>
    <li>python314Packages.wandb.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 122 packages built:</summary>
  <ul>
    <li>python314Packages.altair</li>
    <li>python314Packages.altair.dist</li>
    <li>python314Packages.backtesting</li>
    <li>python314Packages.backtesting.dist</li>
    <li>python314Packages.bokeh</li>
    <li>python314Packages.bokeh.dist</li>
    <li>python314Packages.bumps</li>
    <li>python314Packages.bumps.dist</li>
    <li>python314Packages.chart-studio</li>
    <li>python314Packages.chart-studio.dist</li>
    <li>python314Packages.dash</li>
    <li>python314Packages.dash-bootstrap-components</li>
    <li>python314Packages.dash-bootstrap-components.dist</li>
    <li>python314Packages.dash-bootstrap-templates</li>
    <li>python314Packages.dash-bootstrap-templates.dist</li>
    <li>python314Packages.dash.dist</li>
    <li>python314Packages.experiment-utilities</li>
    <li>python314Packages.experiment-utilities.dist</li>
    <li>python314Packages.explorerscript</li>
    <li>python314Packages.explorerscript.dist</li>
    <li>python314Packages.extra-streamlit-components</li>
    <li>python314Packages.extra-streamlit-components.dist</li>
    <li>python314Packages.formulaic</li>
    <li>python314Packages.formulaic.dist</li>
    <li>python314Packages.gradio</li>
    <li>python314Packages.gradio.dist</li>
    <li>python314Packages.great-tables</li>
    <li>python314Packages.great-tables.dist</li>
    <li>python314Packages.holistic-trace-analysis</li>
    <li>python314Packages.holistic-trace-analysis.dist</li>
    <li>python314Packages.holoviews</li>
    <li>python314Packages.holoviews.dist</li>
    <li>python314Packages.hvplot</li>
    <li>python314Packages.hvplot.dist</li>
    <li>python314Packages.ibis-framework</li>
    <li>python314Packages.ibis-framework.dist</li>
    <li>python314Packages.igraph</li>
    <li>python314Packages.igraph.dist</li>
    <li>python314Packages.intake</li>
    <li>python314Packages.intake.dist</li>
    <li>python314Packages.iplotx</li>
    <li>python314Packages.iplotx.dist</li>
    <li>python314Packages.itables</li>
    <li>python314Packages.itables.dist</li>
    <li>python314Packages.kmapper</li>
    <li>python314Packages.kmapper.dist</li>
    <li>python314Packages.layoutparser</li>
    <li>python314Packages.layoutparser.dist</li>
    <li>python314Packages.leidenalg</li>
    <li>python314Packages.leidenalg.dist</li>
    <li>python314Packages.lida</li>
    <li>python314Packages.lida.dist</li>
    <li>python314Packages.lifelines</li>
    <li>python314Packages.lifelines.dist</li>
    <li>python314Packages.linearmodels</li>
    <li>python314Packages.linearmodels.dist</li>
    <li>python314Packages.livelossplot</li>
    <li>python314Packages.livelossplot.dist</li>
    <li>python314Packages.marimo</li>
    <li>python314Packages.marimo.dist</li>
    <li>python314Packages.meshtastic</li>
    <li>python314Packages.meshtastic.dist</li>
    <li>python314Packages.msticpy</li>
    <li>python314Packages.msticpy.dist</li>
    <li>python314Packages.narwhals</li>
    <li>python314Packages.narwhals.dist</li>
    <li>python314Packages.nbsmoke</li>
    <li>python314Packages.nbsmoke.dist</li>
    <li>python314Packages.neurokit2</li>
    <li>python314Packages.neurokit2.dist</li>
    <li>python314Packages.niaarm</li>
    <li>python314Packages.niaarm.dist</li>
    <li>python314Packages.pandantic</li>
    <li>python314Packages.pandantic.dist</li>
    <li>python314Packages.pandas-stubs</li>
    <li>python314Packages.pandas-stubs.dist</li>
    <li>python314Packages.pandera</li>
    <li>python314Packages.pandera.dist</li>
    <li>python314Packages.panel</li>
    <li>python314Packages.panel.dist</li>
    <li>python314Packages.pdfplumber</li>
    <li>python314Packages.pdfplumber.dist</li>
    <li>python314Packages.plotly</li>
    <li>python314Packages.plotly.dist</li>
    <li>python314Packages.pyreadstat</li>
    <li>python314Packages.pyreadstat.dist</li>
    <li>python314Packages.reflex</li>
    <li>python314Packages.reflex-chakra</li>
    <li>python314Packages.reflex-chakra.dist</li>
    <li>python314Packages.reflex.dist</li>
    <li>python314Packages.sasmodels</li>
    <li>python314Packages.sasmodels.dist</li>
    <li>python314Packages.shiny</li>
    <li>python314Packages.shiny.dist</li>
    <li>python314Packages.skops</li>
    <li>python314Packages.skops.dist</li>
    <li>python314Packages.smolagents</li>
    <li>python314Packages.smolagents.dist</li>
    <li>python314Packages.st-pages</li>
    <li>python314Packages.st-pages.dist</li>
    <li>python314Packages.streamlit</li>
    <li>python314Packages.streamlit-avatar</li>
    <li>python314Packages.streamlit-avatar.dist</li>
    <li>python314Packages.streamlit-card</li>
    <li>python314Packages.streamlit-card.dist</li>
    <li>python314Packages.streamlit-echarts</li>
    <li>python314Packages.streamlit-echarts.dist</li>
    <li>python314Packages.streamlit-folium</li>
    <li>python314Packages.streamlit-folium.dist</li>
    <li>python314Packages.streamlit-kpi-card</li>
    <li>python314Packages.streamlit-kpi-card.dist</li>
    <li>python314Packages.streamlit-notify</li>
    <li>python314Packages.streamlit-notify.dist</li>
    <li>python314Packages.streamlit.dist</li>
    <li>python314Packages.synergy</li>
    <li>python314Packages.synergy.dist</li>
    <li>python314Packages.txtai</li>
    <li>python314Packages.txtai.dist</li>
    <li>python314Packages.unstructured-inference</li>
    <li>python314Packages.unstructured-inference.dist</li>
    <li>python314Packages.vega</li>
    <li>python314Packages.vega.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python314Packages.wandb</summary>
<pre>Setting COVERAGE_FILE to /build/source/.coverage
=============================== warnings summary ===============================
../../nix/store/v0fq3sahvxzn25zg8clv3lcnxyq1d21s-python3.14-pydantic-2.12.4/lib/python3.14/site-packages/pydantic/_internal/_generate_schema.py:2249: 129 warnings
  /nix/store/v0fq3sahvxzn25zg8clv3lcnxyq1d21s-python3.14-pydantic-2.12.4/lib/python3.14/site-packages/pydantic/_internal/_generate_schema.py:2249: UnsupportedFieldAttributeWarning: The &#x27;repr&#x27; attribute with value False was provided to the `Field()` function, which has no effect in the context it was used. &#x27;repr&#x27; is field-specific metadata, and can only be attached to a model field using `Annotated` metadata or by assignment. This may have happened because an `Annotated` type alias using the `type` statement was used, or if the `Field()` function was attached to a single member of a union type.
    warnings.warn(

../../nix/store/v0fq3sahvxzn25zg8clv3lcnxyq1d21s-python3.14-pydantic-2.12.4/lib/python3.14/site-packages/pydantic/_internal/_generate_schema.py:2249: 129 warnings
  /nix/store/v0fq3sahvxzn25zg8clv3lcnxyq1d21s-python3.14-pydantic-2.12.4/lib/python3.14/site-packages/pydantic/_internal/_generate_schema.py:2249: UnsupportedFieldAttributeWarning: The &#x27;frozen&#x27; attribute with value True was provided to the `Field()` function, which has no effect in the context it was used. &#x27;frozen&#x27; is field-specific metadata, and can only be attached to a model field using `Annotated` metadata or by assignment. This may have happened because an `Annotated` type alias using the `type` statement was used, or if the `Field()` function was attached to a single member of a union type.
    warnings.warn(

tests/unit_tests/test_wandb_settings_old.py::test__global_path_env_var_exists_and_is_writable
  /build/source/tests/unit_tests/test_wandb_settings_old.py:33: PytestWarning: Value of environment variable WANDB_CONFIG_DIR type should be str, but got PosixPath(&#x27;/build/pytest-of-nixbld/pytest-0/popen-gw16/test__global_path_env_var_exis0/config&#x27;) (type: PosixPath); converted to str implicitly
    monkeypatch.setenv(&quot;WANDB_CONFIG_DIR&quot;, config_dir)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/unit_tests/test_torch.py::test_watch_parameters_torch_jit[all] - AttributeError: &#x27;Linear&#x27; object has no attribute &#x27;__annotations__&#x27;. Did you mean: &#x27;__annotate_func__&#x27;?
FAILED tests/unit_tests/test_torch.py::test_watch_parameters_torch_jit[parameters] - AttributeError: &#x27;Linear&#x27; object has no attribute &#x27;__annotations__&#x27;. Did you mean: &#x27;__annotate_func__&#x27;?
FAILED tests/unit_tests/test_torch.py::test_watch_graph_torch_jit - AttributeError: &#x27;Net&#x27; object has no attribute &#x27;__annotations__&#x27;. Did you mean: &#x27;__annotate_func__&#x27;?
= 3 failed, 1832 passed, 142 skipped, 3 xfailed, 5 xpassed, 259 warnings in 232.65s (0:03:52) =</pre>
</details>
</details>

--- a/pkgs/development/python-modules/ibis-framework/default.nix
+++ b/pkgs/development/python-modules/ibis-framework/default.nix
@@ -97,7 +97,7 @@ let
   };
 in
 
-buildPythonPackage rec {
+buildPythonPackage (finalAttrs: {
   pname = "ibis-framework";
   version = "11.0.0";
   pyproject = true;
@@ -105,7 +105,7 @@ buildPythonPackage rec {
   src = fetchFromGitHub {
     owner = "ibis-project";
     repo = "ibis";
-    tag = version;
+    tag = finalAttrs.version;
     hash = "sha256-hf5guWeX9WQbKaNrs7ALwwDxV1Rgeb5Z0PedTQ4P7S0=";
   };
 
@@ -139,7 +139,7 @@ buildPythonPackage rec {
     pytest-xdist
     writableTmpDirAsHomeHook
   ]
-  ++ lib.concatMap (name: optional-dependencies.${name}) testBackends;
+  ++ lib.concatMap (name: finalAttrs.passthru.optional-dependencies.${name}) testBackends;
 
   pytestFlags = [
     "--benchmark-disable"
@@ -372,11 +372,11 @@ buildPythonPackage rec {
   meta = {
     description = "Productivity-centric Python Big Data Framework";
     homepage = "https://github.com/ibis-project/ibis";
-    changelog = "https://github.com/ibis-project/ibis/blob/${src.tag}/docs/release_notes.md";
+    changelog = "https://github.com/ibis-project/ibis/blob/${finalAttrs.src.tag}/docs/release_notes.md";
     license = lib.licenses.asl20;
     maintainers = with lib.maintainers; [
       cpcloud
       sarahec
     ];
   };
-}
+})

--- a/pkgs/development/python-modules/ibis-framework/default.nix
+++ b/pkgs/development/python-modules/ibis-framework/default.nix
@@ -186,6 +186,7 @@ buildPythonPackage (finalAttrs: {
   ]
   ++ lib.optionals (pythonAtLeast "3.14") [
     # ExceptionGroup: multiple unraisable exception warnings (4 sub-exceptions)
+    "test_non_roundtripable_str_type"
     "test_parse_dtype_roundtrip"
 
     # AssertionError: value does not match the expected value in snapshot ...

